executor/common_linux.h

```diff
@@ -2678,19 +2678,16 @@ static long syz_emit_vhci(volatile long a0, volatile long a1)
 static long syz_genetlink_get_family_id(volatile long name, volatile long sock_arg)
 {
 	debug("syz_genetlink_get_family_id(%s, %d)\n", (char*)name, (int)sock_arg);
-	// We can't trust the socket passed by the fuzzer, it may be not a netlink at all.
-	bool dofail = false;
 	int fd = sock_arg;
 	if (fd < 0) {
-		dofail = true;
 		fd = socket(AF_NETLINK, SOCK_RAW, NETLINK_GENERIC);
 		if (fd == -1) {
 			debug("syz_genetlink_get_family_id: socket failed: %d\n", errno);
 			return -1;
 		}
 	}
 	struct nlmsg nlmsg_tmp;
-	int ret = netlink_query_family_id(&nlmsg_tmp, fd, (char*)name, dofail);
+	int ret = netlink_query_family_id(&nlmsg_tmp, fd, (char*)name, false);
```

Comment:
If this call with a locally-created socket can fail, why other similar cases can't fail? If we expect that such calls with a locally-created socket can fail, shouldn't we remove this 'dofail' argument entirely?

Reply:
If other dofail=true cases happen during sandboxing, then I guess it is reasonable and makes sense.